### PR TITLE
Continue running tests even if previous unit tests failed

### PR DESCRIPTION
**Pull request by asenyaev**
This PR fixes https://github.com/opencv/ci-gha-workflow/issues/7 for current workflows.

### CODE REVIEW

**Review by fengyuentau — CHANGES_REQUESTED** [review]
Can you confirm whether this condition `if: ${{ always() && steps.build-opencv-contrib.outcome == 'success' }}` can stop a step if build is failed or the workflow is cancelled?

File: .github/workflows/OCV-Contrib-PR-3.4-ARM64.yaml [hunk 1]
```diff
@@ -108,136 +108,198 @@ jobs:
...
       run: |
         cd /opencv/modules/python/test
         python3 ./test.py --repo ../../../ -v
     - name: Java
+      if: ${{ always() && steps.build-opencv-contrib.outcome == 'success' }}
       run: cd /build && xvfb-run -a python3 /opencv/modules/ts/misc/run.py . -a -t java
     - name: Save Unit Test Results
       uses: actions/upload-artifact@v3
-      if: always()
+      if: ${{ success() && steps.build-opencv-contrib.outcome == 'success' }}
```
[the remark this answers is not in the document]
Comment (asenyaev) [reply .github/workflows/OCV-Contrib-PR-3.4-ARM64.yaml]: Agree, added an id for java test and changed the if condition.

**Comment by asenyaev** [comment]
> Can you confirm whether this condition `if: ${{ always() && steps.build-opencv-contrib.outcome == 'success' }}` can stop a step if build is failed or the workflow is cancelled?

Yes, the example job is [here](https://github.com/opencv/ci-gha-workflow/runs/6708090236?check_suite_focus=true).

**Review by fengyuentau — APPROVED** [review]
👍

**Pull request merged**